ONSdigital/ras-secure-message
# database error handling

# What and why?
Adds error handling around database errors and tests around this. Also updated the GHA workflow with the new `setup-gcloud` and replaces how we create the release
# How to test?
I haven't found a way to test this easily when deployed through spinnaker as other services fall over when there is a database connection issue (I cause this by deleting the postgres configmap). Instead I deploy the secure message PR locally and either insert `raise SQLAlchemyError` where it's being tested or then stop the local postgres deployment
- update uaa client secret so that secure message can connect to uaa on startup
- run local postgres, securemessage, rOps and/or frontstage
- make a SQLAlchemyError happen
- make sure that the whole stack trace is not printed to the logs
# Jira
[RAS-1031](https://jira.ons.gov.uk/browse/RAS-1031)
